src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,48 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     *
+     * @since 4.13
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            if (throwableClass.isInstance(t)) {
+                return throwableClass.cast(t);
```

Comment:
I've already added the `assertThrows` variant, I'm just working on some minor feedback from a few weeks ago now.

"I would just do a simple cast here; you know it's safe because you just checked, and it's (slightly) more efficient and just as clear."

Believe it or not, when I use a simple cast I get a compiler warning about an unchecked cast. It goes away when I use the `cast` method. I'm not sure what to do with that information. Should I go with a simple cast anyway? If so, should I add a `@SuppressWarnings` annotation?